notation/verifier.go

```diff
@@ -97,12 +98,16 @@ func (v *Verifier) Verify(ctx context.Context, opts *ratify.VerifyOptions) (*rat
 		return nil, fmt.Errorf("failed to parse subject reference: %w", err)
 	}
 
-	signatureDesc, err := v.getSignatureBlobDesc(ctx, opts.Store, subjectRef, opts.SubjectDescriptor)
+	signatureDesc, err := v.getSignatureBlobDesc(ctx, opts.Store, subjectRef, opts.ArtifactDescriptor)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get signature blob descriptor: %w", err)
 	}
 
-	signatureBlob, err := opts.Store.FetchBlobContent(ctx, subjectRef.Repository, signatureDesc)
+	repo := subjectRef.Repository
+	if subjectRef.Registry != "" {
```

Comment:
This is always true by line 96

Reply:
synced offine, will update the VerifyOptions so that we can get the full repository path instead of constructing it.